chrome/content/zotero/xpcom/integration.js

```diff
@@ -2794,6 +2799,13 @@ Zotero.Integration.Progress = class {
 			await Zotero.Promise.delay(300);
 		}
 		this.window.close();
+		// macOS switches focus to last frontmost window of that application when a window is closed,
+		// instead of switching to the last frontmost window from any application
+		if (Zotero.isMac) {
+			Zotero.Utilities.Internal.sendToBack();
+			// Wait a bit to ensure that other window activations are executed in proper order
+			await Zotero.Promise.delay(100);
+		}
```

Comment:
> In a new doc with Word 2011, I click Add Citation, Zotero and the doc prefs come to the front, I choose APA and OK, and then Zotero becomes hidden — i.e., actually grayed out in the Dock. So I'm left with just Word and "{Citation}".

I added this progress meter as an independent Zotero window for various integration operations that used to have no progress indication before. E.g. The progress meter window is displayed when retrieving fields from the document, before showing the document preferences window. The problem that surfaced here

> In this version, using LibreOffice on macOS, if I go to add a citation it brings Zotero to the front and shows the Quick Format bar on top of that. If I switch back to master, the Quick Format bar shows on top of LibreOffice. (Zotero sometimes flashes to the front but it generally stays behind LibreOffice.)

is rooted in how macOS handles closed application windows (described in the code comment above).

We can hide the Zotero window using AppleScript, but then the problem is that if AppleScript takes a long time to execute, the next window to display (the citation, preferences, bibl edit windows) may get hidden too. We can either keep the Zotero window on top while displaying the citation dialog, or we can keep on increasing this delay above (currently 100ms). We can probably go up to 500ms without a significant usability decrease, but if that is not enough, I am out of suggestions.

Reply:
I don't think we should do anything that relies on a specific delay. If we can't do this properly, we should just drop the extra progress window on macOS.

Reply:
I don't think bringing Zotero to the front with a progress bar would be the end of the world for Google Docs, but we shouldn't regress for everything else.

I'm not clear on the consequences of increasing the delay, but we definitely can't risk leaving Zotero in a hidden state with "{Citation}" displayed, and we don't want to actually slow things down further if that's a possibility.

I also still think that "{Updating}" would be clearer, and possibly obviate the need for a progress bar at all, at least in Word/LibreOffice.

Reply:
> I don't think bringing Zotero to the front with a progress bar would be the end of the world for Google Docs, but we shouldn't regress for everything else.

I think this is the correct solution.

Reply:
Updated